File: Python/Dosen.py
# Saya Hakasa Putri mengerjakan LP3 DPBO 2023 C2 
# dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan 
# kecurangan seperti yang telah dispesifikasikan. Aamiin 


#melakukan import class
from SivitasAkademik import SivitasAkademik
from Prodi import Prodi

class Dosen(SivitasAkademik):

    # atribut private
    __NIP = ""
    __Fakultas = ""
    __PendTerakhir = ""
    __Keahlian = ""

    def __init__(self): #Constructor

        # atribut
        self. __NIP = ""
        self.__Fakultas = ""
        self.__PendTerakhir = ""
        self.__Keahlian = ""

    # setter getter
    def getNIP(self):
        return self.NIP
    
    def setNIP(self, NIP):
        self.NIP = NIP

    def getFakultas(self):
        return self.Fakultas
    
    def setFakultas(self, Fakultas):
        self.Fakultas = Fakultas
    
    def getPendTerakhir(self):
        return self.PendTerakhir
    
    def setPendTerakhir(self, PendTerakhir):
        self.PendTerakhir = PendTerakhir
    
    def getKeahlian(self):
        return self.Keahlian
    
    def setKeahlian(self, Keahlian):
        self.Keahlian = Keahlian

    def getProdi(self):
        return self.Prodi
    
    def setProdi(self, Prodi):
        self.Prodi = Prodi


    # untuk print atribut kelas dosen
    def getInfoDosen(self):

        print("-------------------------------------------")
        print("yang merupakan seorang dosen dengan")
        print("-------------------------------------------")
        print("NIP            : " + self.getNIP())
        print("Fakultas       : " + self.getFakultas())
        print("Pend Terakhir  : " + self.getPendTerakhir())
        print("Keahlian       : " + self.getKeahlian())
        self.Prodi.getInfoProdi()

     
    

File: C++/Program/Dosen.cpp
// Saya Hakasa Putri mengerjakan LP3 DPBO 2023 C2
// dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan
// kecurangan seperti yang telah dispesifikasikan. Aamiin

#include <iostream>


using namespace std;

class Dosen {
private:
string NIP;
string Fakultas;
string PendTerakhir;
string Keahlian;
Prodi prodi;

public:
//Constructor
Dosen(){
NIP = "";
Fakultas = "";
PendTerakhir = "";
Keahlian = "";
}

// setter getter
string getNIP(){
    return NIP;
}

void setNIP(string nip){
    NIP = nip;
}

string getFakultas(){
    return Fakultas;
}

void setFakultas(string fakultas){
    Fakultas = fakultas;
}

string getPendTerakhir(){
    return PendTerakhir;
}

void setPendTerakhir(string pendTerakhir){
    PendTerakhir = pendTerakhir;
}

string getKeahlian(){
    return Keahlian;
}

void setKeahlian(string keahlian){
    Keahlian = keahlian;
}

Prodi getProdi(){
    return prodi;
}

void setProdi(Prodi prodi){
    this->prodi = prodi;
}

// untuk print atribut kelas dosen
void getInfoDosen(){

    cout << "-------------------------------------------" << endl;
    cout << "yang merupakan seorang dosen dengan" << endl;
    cout << "-------------------------------------------" << endl;
    cout << "NIP            : " << getNIP() << endl;
    cout << "Fakultas       : " << getFakultas() << endl;
    cout << "Pend Terakhir  : " << getPendTerakhir() << endl;
    cout << "Keahlian      : " << getKeahlian() << endl;

    prodi.getInfoProdi();

    }
};
